### ISSUE TITLE
Cannot read properties of null (reading 'isCE') since 3.11.0. Same code is ok on 3.10.3 and below

### ISSUE THREAD

**Issue by marcoborsoi** [nuxt/nuxt#28468]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.11.0
- Nuxt Version:     3.11.0
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  pnpm@8.14.2
- Builder:          -
- User Config:      devtools, modules
- Runtime Modules:  @myorg/shared-ui-nuxt@1.0.0
- Build Modules:    -
------------------------------

### Reproduction

https://github.com/marcoborsoi/nuxt-monorepo-poc

### Describe the bug

I have a monorepo containing a Nuxt application (`nuxt-app`), a lib of Vue components (`shared-ui`) and a Nuxt module (`shared-ui-nuxt`) responsible for registering `shared-ui` components and it's CSS in a Nuxt app. The application is deployed in Netlify.

The solution works fine until Nuxt@3.10.3. Since 3.11.0, a Nuxt error page with the message "500 - Cannot read properties of null (reading 'isCE')" is been displayed. The Netlify server logs shows "WARN   [Vue warn]: resolveComponent can only be used in render() or setup()."

- `shared-ui-nuxt` was largely inspired by Vuestic Nuxt module: https://github.com/epicmaxco/vuestic-ui/tree/develop/packages/nuxt
- `shared-ui`is a dependency of `shared-ui-nuxt`.
- `shared-ui` and `shared-ui-nuxt` are dependencies of `nuxt-app`.
- `shared-ui` exports a Vue component called `VTag` and it can be added to `nuxt-app` via `shared-ui-nuxt` in this way:

```ts
// libs/shared/ui-nuxt/module.ts

// `shared-ui-nuxt` contains it's own set of components in `runtime/components`. This works.
addComponentsDir({
  path: resolver.resolve('../src/runtime/components'),
  pathPrefix: false,
  global: true,
})

// This works on nuxt@3.10.3 and below, but it does not work on nuxt@3.11.0 and above.
addComponent({
  name: 'VSpan',
  export: 'VSpan',
  filePath: '@myorg/shared-ui',
})
```

- `shared-ui-nuxt` is registered in `nuxt-app`:
```ts
// apps/nuxt-app/nuxt.config.ts

export default defineNuxtConfig({
  devtools: {
    enabled: true,
  },

  modules: ['@myorg/shared-ui-nuxt'],
})
```

![Screenshot 2024-08-08 at 9 31 46 AM](https://github.com/user-attachments/assets/70b1d8cd-05d1-4040-b56b-778f31ee4670)

**Important:** the issue does not occur when running in development mode neither with `nuxt preview`, but it fails once running in Netlify.

The reproduction link is a much simplified version of the commercial solution currently in production running on 3.10.3 (whose code I am not allowed to share unfortunately), but it shows the issue I am facing in a clearer manner. By changing Nuxt from 3.11.0 (and above) to 3.10.3 version makes the application work and we can see the `VTag` component exposed by the module being correctly rendered in `nuxt-app`.

I wonder what changed in 3.11.0 that could lead to this issue and, if I am making a mistake in the way the `shared-ui` components should be exposed, why it worked until 3.10.3.

Please let me know if any other information is necessary.

Thank you for your time!

### Logs

```shell-script
// Netlify server logs:

WARN   [Vue warn]: resolveComponent can only be used in render() or setup().
```

**Comment by gianluigibitboss** [comment]
Did you resolve this? Stuck as well behaviour is super weird.
I have  a module whose playground works flawlessly.
Components imported from the library work with no issues.
Components added through addComponent thro the same error as you have.
It's been two days i've been banging my head to get it to work.

**Comment by marcoborsoi** [comment]
> Did you resolve this? Stuck as well behaviour is super weird. I have a module whose playground works flawlessly. Components imported from the library work with no issues. Components added through addComponent thro the same error as you have. It's been two days i've been banging my head to get it to work.

[gianluigibitboss](https://github.com/gianluigibitboss) In the case of the monorepo I'm working on, `vue` and `vue-router` were `devDependencies` of the workspace along with `nuxt`. I only got it working after setting `vue` and `vue-router`  as `dependency` of the Nuxt application itself, and not to the workspace.

In the case of my project, we reviewed the monorepo structure as well and we decided to migrate all the components previously exported by the lib of pure Vue components (`shared-ui`) to the nuxt module itself (`shared-ui-nuxt`), so all the components are registered in the target Nuxt app via `addComponentsDir`.

I hope this can shed some light for you.
